baselines/fed_debug/README.md

```diff
@@ -0,0 +1,164 @@
+
+
+**<u>Application of FedDebug</u>:** We used FedDebug to detect backdoor attacks in Federated Learning, resulting in [Fed-Defender](https://dl.acm.org/doi/10.1145/3617574.3617858). The code is implemented using the Flower Framework in [this repository](https://github.com/warisgill/FedDefender). We plan to adapt Fed-Defender to Flower baseline guidelines soon.
+
+
+
+## About this baseline
+**What's implemented:** 
+
+FedDebug is a systematic malicious client(s) localization framework designed to advance debugging in Federated Learning (FL). It enables interactive debugging of real-time collaborative training and automatically identifies clients responsible for lowering global model performance without requiring testing data or labels.
+
+This repository implements the FedDebug technique of localizing malicious client(s) in a generic way, allowing it to be used with various fusion techniques and CNN architectures. Specifically, it replicates the results presented in Table 2 and Figure 10 of the original paper. These results demonstrate FedDebug's ability to localize multiple faulty clients and its performance under different neuron activation thresholds. You can find the original code of FedDebug [here](https://github.com/SEED-VT/FedDebug).
+
+
+
+
+
+
+**Datasets:** This baseline uses two datasets:
+1. CIFAR-10: 50,000 training and 10,000 testing 32x32 RGB images across 10 classes.
+2. FEMNIST: Over 340,000 training and 40,000 testing 28x28 grayscale images across 10 classes.
+
+**Hardware Setup:**
+These experiments were run on a machine with 8 CPU cores and an Nvidia Tesla P40 GPU. You can run these experiments even on CPU but it will take time.
+
+**Contributors:** Waris Gill ([GitHub Profile](https://github.com/warisgill))
+
+## Experimental Setup
+
+**Task:** Image classification, Malicious/Faulty Client (s) Removal, Debugging and Testing
+
+**Models:** This baseline implements two CNN architectures:
+1. ResNet
+2. DenseNet
+3. VGG
+
+**Dataset:** The datasets are partitioned among clients, and each client participates in the training (cross-silo). However, you can easily extend the code to work in cross-device settings. The non-IID is based on quantity-based imbalance ([niid_bench](https://arxiv.org/abs/2102.02079)).
+
+| Dataset | #classes | #clients | partitioning method |
+| :------- | :------: | :------: | :-----------------: |
+| CIFAR-10 | 10 | 30, 50 | IID and Non-IID |
+| FEMNIST | 10 | 30, 50 | IID and Non-IID |
+
+  
+**Training Hyperparameters:**
+
+Default training hyperparameters are in `conf/base.yaml`.
+
+## Environment Setup
```

Comment:
It should be `3.10.something` in your `pyproject.toml`, else the CI tests won't pass.